ocaml/xapi/xapi_observer.ml

```diff
@@ -160,7 +338,7 @@ let assert_valid_components components =
 
 let assert_valid_endpoints endpoints =
   let validate_endpoint = function
-    | "bugtool" ->
+    | str when Tracing.(endpoint_of_string str = Bugtool) ->
```

Comment:
minor style point: we have several places where we want to use the string "bugtool", maybe instead of doing the convertion each time, we could define an Endpoint module and then hardcode a "bugtool" variable there (and then export it)?